tests/swarm-deploy/test_swarm_runs.py

```diff
@@ -84,12 +84,16 @@ def core_services_running(docker_client: DockerClient) -> List[Service]:
 
 def test_all_services_up(core_services_running: str, make_up_prod: Dict):
     running_services = sorted([s.name for s in core_services_running])
+    # remove sidecar_gpu because its a special case for development
+    running_services = [x for x in running_services if not x.endswith("sidecar_gpu")]
     assert running_services == stack_service_names
 
     expected = [
         f"{stack_name}_{service_name}"
         for service_name in make_up_prod[stack_name]["services"].keys()
     ]
+    # remove sidecar_gpu because its a special case for development
+    expected = [x for x in expected if not x.endswith("sidecar_gpu")]

```

Comment:
since you modified the base docker-compose.yml file it is not only in development that the sidecar_gpu exists. but everywhere including in ops. So now you will probably need to adjust in osparc-ops as well. Therefore I would try to find another solution.
in case it should be only in development, then add the service to docker-compose.devel only or in docker-compose.local only but not in docker-compose.yml.

still, I think in case this is for local development, I would try to keep only one sidecar service and allow it to process both CPU and GPU tasks in case a GPU is available.

Reply:
I have tried to keep only one container, but with how celery works I need to separate apps to handle this. I think the best solution still remains to keep 2 separate containers.

To avoid further issues, moving from `docker-compose.yml` -> `docker-compose.devel.yml`. Turns out this is the correct place to place it.

Reply:
maybe also add a short NOTE in the docker-compose.devel just to indicate the reason

Reply:
Done and also fixed an issue where all sidecars were schedules as CPU on production/staging environments